rcl/src/rcl/arguments.c

```diff
@@ -958,6 +958,7 @@ rcl_arguments_copy(
   args_out->impl->parameter_overrides = NULL;
   args_out->impl->parameter_files = NULL;
   args_out->impl->num_param_files_args = 0;
+  args_out->impl->external_log_config_file = NULL;
```

Comment:
So this is basically duplicating the same code that is at https://github.com/ros2/rcl/blob/7537f8304493e9b9568748b5601f5c1b7fa463f0/rcl/src/rcl/arguments.c#L330 (which is why it was probably missed).

I'm going to suggest that instead of adding this line, we add in a method that is something like `rcl_allocate_and_zero`, which will both do the allocation and setting all of these to default values.  That way, we can't forget something like this in the future.

Reply:
I have the same opinion and do it (183ad826d14e7257d72913bfee36b6360602176d).

Some variables are set in rcl_parse_arguments() but not in rcl_arguments_copy() such as
- log_level
- log_stdout_disabled
- log_rosout_disabled
- log_ext_lib_disabled

Do you know they are only forgotten or there is some reason?
I left it as is because I don't understand rcl_arguments_copy use case in detail.

Reply:
>     * 
> 
> 
> Do you know they are only forgotten or there is some reason?
> I left it as is because I don't understand rcl_arguments_copy use case in detail.

It's not entirely clear to me, but I believe we want to copy them as well.

Reply:
@clalancette Thank you for comment. I fixed.